app/src/main/java/com/example/mystore/ui/products/ProductDetailsScreen.kt

```diff
@@ -44,15 +48,33 @@ import com.ramcosta.composedestinations.navigation.DestinationsNavigator
 
 @Destination
 @Composable
-fun ProductDetailsScreen(destinationsNavigator: DestinationsNavigator, product: ProductModel) {
+fun ProductDetailsScreen(
+    destinationsNavigator: DestinationsNavigator,
+    product: ProductModel
```

Comment:
`product` belongs in the view model. Use `SavedStateHandle` described on the last lecture to provide it directly to the view model, without it passing through the composable at all.